src/ApiHandler.ts

```diff
@@ -139,6 +125,69 @@ export default class ApiHandler {
 			}
 		}
 
+		const byteFee = api.consts.transactionPayment.transactionByteFee;
+		const extrinsicBaseWeight = api.consts.system.extrinsicBaseWeight;
+		const coefficients = api.consts.transactionPayment.weightToFee[0];
+		const coeffFrac = coefficients.coeff_frac;
+		const coeffInt = coefficients.coeff_integer;
+		
+		// TODO: generalize for terms that may have more degrees & may be negative
+		const degree = coefficients.degree;
+		const negative = coefficients.negative;
+
+		for (let idx = 0; idx < block.extrinsics.length; ++idx) {
+			const extrinsic = block.extrinsics[idx];
+			if (!extrinsic.isSigned) {
+				continue;
+			}
+
+			try {
+				if (api.runtimeVersion.specName.toString() === 'kusama') {
+					extrinsics[idx].info = await api.rpc.payment.queryInfo(extrinsic.toHex(), parentHash);
+				} else {
+					const parsedXt = extrinsics[idx];
+					const xtEvents = parsedXt.events;
+					const success = 'system.ExtrinsicSuccess';
+					const failure = 'system.ExtrinsicFailed';
+					const completedEvent = xtEvents.find((event) => event.method === success || event.method === failure);
+					if (!completedEvent) {
+						parsedXt.info = { error: 'Unable to fetch fee info' };
+						continue;
+					}
+
+					const completedData = completedEvent.data;
+					if (!completedData) {
+						parsedXt.info = { error: 'Unable to fetch fee info' };
+						continue;
+					}
+
+					const weightInfo = completedData[completedData.length - 1] as DispatchInfo;
+					if (!weightInfo.weight) {
+						parsedXt.info = { error: 'Unable to fetch fee info' };
+						continue;
+					}
+
+					const weight = weightInfo.weight;
+					const encodedLength = extrinsic.encodedLength;
+					const multiplier = await api.query.transactionPayment.nextFeeMultiplier.at(parentHash);
+
+					const multiplied = multiplier.mul(weight.mul(coeffFrac)
+                                       .add(weight.mul(coeffInt))
+                                       .add(byteFee).muln(encodedLength));
+
+					parsedXt.info = {
+						weight: weight.toString(),
+						class: weightInfo.class.toString(),
+						partialFee: multiplied.add(extrinsicBaseWeight.mul(coeffFrac))
```

Comment:
Dispatch info comes out in the ExtrinsicSuccess event, we are matching the data format of `queryInfo`. It's not a very useful format, but it is what users are set up to parse, so we are forcing the new format into the old to avoid last-minute breaking changes. But, once Kusama is upgraded and things are a bit more stable, we'll make a better interface with plenty of warning to users about the breaking change in response.

Reply:
`query_info` and `partial_fee` are probably the two worst named things in Substrate.

Reply:
I just stumbled upon `RuntimeDispatchInfo` in the Substrate codebase, sorry, my response was nonsense.